elastic/kibana
# [Mappings Editor] Add support for subobjects parameter

Fixes https://github.com/elastic/kibana/issues/183714

## Summary

This PR adds support for the `subobjects` mappings parameter and mappings configuration under "Advanced options" in the Mappings editor UI.


<img width="1491" alt="Screenshot 2024-06-06 at 15 54 11" src="https://github.com/elastic/kibana/assets/59341489/1e5afa03-c49a-4bd0-a9aa-d7c80d625826">


<img width="1491" alt="Screenshot 2024-06-06 at 15 58 32" src="https://github.com/elastic/kibana/assets/59341489/96e566f5-7fbc-4909-922d-d977dd31f23a">


**How to test:**

1. Start Es and Kibana
2. Go to Index Management -> Index Templates/Component Templates and start creating a index/component template.
3. Go to the Mappings step and create a field of type `object`. Edit this field and toggle the `subobjects` field.
4. Go to the "Advanced options" tab and toggle the `subobjects` field.
5. Go to the last Review step and verify that the preview of the template contains the correct `subobjects` value.
6. Save the template and then start editing it. Verify that the `subobject` field is correctly populated in the form.

Test loading a JSON:
1. Start Es and Kibana
2. Go to Index Management -> Index Templates/Component Templates and start creating a index/component template.
7. At the Mappings step, click on the "Load JSON" button.
8. Add the following mappings object (taken from the first example in [Es docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/subobjects.html#subobjects)):
```
{
  "properties": {
    "metrics": {
      "type":  "object",
      "subobjects": false 
    }
  }
}
```
9. Save and go to the Mappings editor - verify that the mappings field is correctly displayed in the editor with the correct `subobjects` field.
10. If creating an index template, click the "Preview index template" button and verify that the `subobjects` parameter is displayed.
11. Finish creating the template and verify it has the `subobjects` parameter in the `mappings` property.
12. Test also loading a JSON with a `subobject` mapping configuration:
```
{
  "subobjects": false 
}
```
Then verify this mappings configuration is correctly applied to the index/component template.
